dbms/src/Flash/Executor/PipelineExecutorStatus.h

```diff
@@ -110,15 +104,20 @@ class PipelineExecutorStatus : private boost::noncopyable
         {
             // If result_handler throws an error, here should notify the query to terminate, and wait for the end of the query.
             onErrorOccurred(std::current_exception());
-            std::unique_lock lock(mu);
-            cv.wait(lock, [&] { return 0 == active_event_count; });
         }
         if (is_timeout)
         {
             LOG_WARNING(log, "wait timeout");
             onErrorOccurred(timeout_err_msg);
             throw Exception(timeout_err_msg);
```

Comment:
How about add a random failpoint here?

Reply:
timeout is only used for unit testing, in order to prevent stuck CI, so there is no need to add failpoint here.

Reply:
ok